libs/klv/src/main/java/org/codice/alliance/libs/klv/LocationKlvProcessor.java

```diff
@@ -85,22 +88,45 @@ public void process(Map<String, KlvHandler> handlers, Metacard metacard,
 
         Integer subsampleCount = (Integer) configuration.get(Configuration.SUBSAMPLE_COUNT);
 
+        tryCornerData(handlers, metacard, subsampleCount);
+
+        if (isLocationNotSet(metacard)) {
+            tryFrameCenterData(handlers, metacard, subsampleCount);
+        }
+
+    }
+
+    private void tryFrameCenterData(Map<String, KlvHandler> handlers, Metacard metacard,
+            Integer subsampleCount) {
+        find(handlers, AttributeNameConstants.FRAME_CENTER).ifPresent(frameCenterHandler -> {
```

Comment:
Looks like `find` cares about both, the attribute name _and_ the handler type. Do you think we should add `Class<T>` as a parameter and do the `instanceof` check inside the `find` method?